p2p/peering.go

```diff
@@ -497,6 +511,51 @@ func (mw *msgWriter) write(ctx context.Context, msg wire.Message, pver uint32) e
 	}
 }
 
+func (rp *RemotePeer) sendInventory(ctx context.Context) error {
+	timeout := func() time.Duration {
+		const d = maxInvTrickleTimeout - minInvTrickleTimeout
+		return minInvTrickleTimeout + rand.Duration(d)
+	}
+	timer := time.NewTimer(timeout())
```

Comment:
What's the point of setting the variable `timer` here? It doesn't look like it's used, and you're already using `timer.Reset(timeout())` below on line 526.

Reply:
Oh I'm sorry, I spoke too soon. You have to wait for `timer.C`, and then you reset it with a new randomized `timeout()`. I misread it as doubling-up the time-duration.